docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+| `host-node`         | Attempts to detect if the environment appears to be a node environment.                                                                                                                                                                                  |
+| `build-time-render` | Statically defined by the build-time rendering system during build-time rendering.                                                                                                                                                                       |
+
+# Build-time Rendering
+
+BTR renders a route to HTML during the build process and in-lines critical CSS and assets needed to display the initial view. This allows Dojo to pre-render the initial HTML used by a route and inject it directly into the page immediately. Skipping the initial render by creating it during the build process gives us many of the same benefits of server side rendering (SSR) such as performance gains and search engine optimization without the complexities of SSR.
+
+## Using BTR
+
+First make sure `index.html` includes a DOM node with an `id` attribute. This node will be used by Dojo's virtual DOM to compare and render the application's HTML. BTR requires this setup so it can render the HTML generated at build time. This creates a very fast and responsive initial rendering of the route. Once the route has had a change to load and render the virtual DOM is able to diff against the BTR.
+
+> index.html
+
+```html
+<!DOCTYPE html>
+<html lang="en-us">
+<head>
+        <title>sample-app</title>
+        <meta name="theme-color" content="#222127">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+</head>
+<body>
+        <div id="root"></div>
+</body>
+</html>
+```
+
+Make sure your application uses the DOM node to mount the renderer.
+
+> main.ts
+
+```ts
+const r = renderer(() => w(App, {}));
+const domNode = document.getElementById('root') as HTMLElement;
+r.mount({ registry, domNode });
+```
+
+Update the `.dojorc` configuration file with the `id` of the root DOM node and routes to render at build time.
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"build-time-render": {
+			"root": "app",
+			"paths": [
+				"#home",
+				{
+					"path": "#comments/9999",
+					"match": ["#comments/.*"]
+				}
+			]
+		}
+	}
+}
+```
+
+This configuration describes two routes. A `home` route and a more complex `comments` route. The `comments` route is a more complex route with parameter data. A `match` is used to make sure that the build-time HTML created for this route is applied to any route that matches the regular expression.
+
+BTR generates a screenshot for each of the paths rendered during the build in the output/info/screenshots directory of your project.
+
+### Hash history
```

Comment:
This header seems confusing to me.

Reply:
Maybe it would be better as a note somewhere? I copied this and am not completely aware of the implications TBH.